src/Microsoft.ML.Data/TrainCatalog.cs

```diff
@@ -638,6 +638,21 @@ internal RankingTrainers(RankingCatalog catalog)
         /// <param name="scoreColumnName">The name of the score column in <paramref name="data"/>.</param>
         /// <returns>The evaluation results for these calibrated outputs.</returns>
         public RankingMetrics Evaluate(IDataView data,
+            string labelColumnName = DefaultColumnNames.Label,
+            string rowGroupColumnName = DefaultColumnNames.GroupId,
+            string scoreColumnName = DefaultColumnNames.Score) => Evaluate(data, null, labelColumnName, rowGroupColumnName, scoreColumnName);
+
+        /// <summary>
+        /// Evaluates scored ranking data.
+        /// </summary>
+        /// <param name="data">The scored data.</param>
+        /// <param name="options">Options to control the evaluation result.</param>
+        /// <param name="labelColumnName">The name of the label column in <paramref name="data"/>.</param>
+        /// <param name="rowGroupColumnName">The name of the groupId column in <paramref name="data"/>.</param>
+        /// <param name="scoreColumnName">The name of the score column in <paramref name="data"/>.</param>
+        /// <returns>The evaluation results for these calibrated outputs.</returns>
+        public RankingMetrics Evaluate(IDataView data,
+            RankingEvaluatorOptions options,
```

Comment:
Can you add at least one new test for this new API?

Reply:
And ensure there is an existing unit test covering the existing MAML interface for `RankingEvaluator`.

It should look about like:
`CV tr=LightGBMRanking eval=RankingEvaluator{t=10} k=2 loader=TextLoader{col=Label:R4:0 col=GroupId:TX:1 col=Features:R4:2-138 header=+} data=MSLR-WEB10K_Fold1.TRAIN.SMALL_10k-rows.tsv xf=HashTransform{col=GroupId} xf=NAHandleTransform{col=Features}`

Specifically the `eval=RankingEvaluator{t=10}` part, where the DCG/NDCG threshold is set.

The output should end in a block of DCG/NDCG@N metrics:
```
OVERALL RESULTS
---------------------------------------
NDCG@1:             51.879160 (3.2943)
NDCG@2:             52.490449 (1.6035)
NDCG@3:             51.704020 (1.7012)
NDCG@4:             51.495099 (0.8726)
NDCG@5:             51.859874 (0.2759)
NDCG@6:             52.278430 (0.2561)
NDCG@7:             52.641604 (0.4342)
NDCG@8:             52.856661 (0.5255)
NDCG@9:             54.075498 (0.5233)
NDCG@10:            54.692374 (0.6690)
DCG@1:              10.778756 (0.1824)
DCG@2:              15.649059 (0.6480)
DCG@3:              18.580052 (0.5112)
DCG@4:              20.808287 (0.9611)
DCG@5:              22.819410 (1.1504)
DCG@6:              24.494010 (1.2360)
DCG@7:              26.108059 (1.6727)
DCG@8:              27.384487 (1.7825)
DCG@9:              29.079234 (1.8749)
DCG@10:             30.308760 (1.9971)

---------------------------------------
Physical memory usage(MB): 123
Virtual memory usage(MB): 5114
8/22/2019 4:56:55 PM	 Time elapsed(s): 5.534
``` 

And it should match the `@N` set in the RankingEvaluator's t=N.